pstatus-graphql-ktor/src/main/kotlin/gov/cdc/ocio/processingstatusapi/mutations/UploadErrorsNotificationSubscriptionMutationService.kt

```diff
@@ -0,0 +1,146 @@
+ */
+@Serializable
+data class UploadErrorsNotificationUnSubscription(val subscriptionId:String)
+
+
+/**
+ * The graphQL mutation service class for upload errors notification subscription/unSubscription
+ */
+
+class UploadErrorsNotificationSubscriptionMutationService : Mutation {
+    private val uploadErrorsNotificationSubscriptionUrl: String = System.getenv("PSTATUS_WORKFLOW_NOTIFICATIONS_BASE_URL")
+    private val serviceUnavailable =
+        "DeadlineCheckSubscription service is unavailable and no connection has been established. Make sure the service is running"
+    private val client = HttpClient {
+        install(ContentNegotiation) {
+            json()
+        }
+        install(Logging) {
+            logger = Logger.DEFAULT
+            level = LogLevel.INFO
+        }
+        install(HttpTimeout) {
+            requestTimeoutMillis = 10000
+            connectTimeoutMillis = 10000
+            socketTimeoutMillis = 10000
+        }
+    }
+
+    /**
+     *  The mutation function which invokes the upload errors notification microservice route to subscribe to it
+     * @param dataStreamId String
+     * @param dataStreamRoute String
+     * @param jurisdiction String
+     * @param daysToRun List<String>
+     * @param deliveryReference String
+     */
+
+    @GraphQLDescription("Subscribe Deadline Check")
+    @Suppress("unused")
+    fun subscribeUploadErrorsNotification(
+        dataStreamId: String,
+        dataStreamRoute: String,
+        jurisdiction: String,
+        daysToRun: List<String>,
+        timeToRun: String,
+        deliveryReference: String
+    ): NotificationSubscriptionResult {
+        val url = "$uploadErrorsNotificationSubscriptionUrl/subscribe/uploadErrorsNotification"
+
+        return runBlocking {
+            try {
+                val response = client.post(url) {
+                    contentType(ContentType.Application.Json)
+                    setBody(
+                        UploadErrorsNotificationSubscription(
+                            dataStreamId,
+                            dataStreamRoute,
+                            jurisdiction,
+                            daysToRun,
+                            timeToRun,
+                            deliveryReference
+                        )
+                    )
+                }
+                return@runBlocking SubscriptionResponse.ProcessNotificationResponse(response)
+            } catch (e: Exception) {
+                if (e.message!!.contains("Status:")) {
+                    SubscriptionResponse.ProcessErrorCodes(url, e, null)
+                }
+                throw Exception(serviceUnavailable)
+            }
+        }
+    }
+
+    /**
+     *  The mutation function which invokes the upload errors in the upload microservice route to unsubscribe
+     * @param subscriptionId String
+     */
+
+    @GraphQLDescription("UnSubscribe Deadline Check")
+    @Suppress("unused")
+    fun unsubscribeUploadErrorsNotification(
```

Comment:
Are we able to consolidate on one unsubscribe like we have in the design?  I thought we'd cover that as part of this work.  Similarly, I thought we were going to consolidate the subscribe mutations as well per the new lucidchart design.